--- /var/www/html/moodle_20160301/mod/quiz/report/attemptsreport.php	2015-03-26 15:49:01.000000000 +0900
+++ /var/www/html/moodle_20160301_analytics/mod/quiz/report/attemptsreport.php	2017-03-07 19:08:08.000000000 +0900
@@ -153,10 +153,10 @@
             $headers[] = '';
         }
         if (!$table->is_downloading()) {
-            $columns[] = 'username';
+            $columns[] = 'userid';
             $headers[] = get_string('name');
         } else {
-            $columns[] = 'username';
+            $columns[] = 'userid';
             $headers[] = get_string('name');
         }
 
