--- /var/www/html/moodle_20160301/mod/quiz/report/attemptsreport_table.php	2015-03-26 15:49:01.000000000 +0900
+++ /var/www/html/moodle_20160301_analytics/mod/quiz/report/attemptsreport_table.php	2017-03-07 19:11:39.000000000 +0900
@@ -163,7 +163,7 @@
      */
     public function col_timestart($attempt) {
         if ($attempt->attempt) {
-            return userdate($attempt->timestart, $this->strtimeformat);
+            return $attempt->timestart;
         } else {
             return  '-';
         }
@@ -176,7 +176,7 @@
      */
     public function col_timefinish($attempt) {
         if ($attempt->attempt && $attempt->timefinish) {
-            return userdate($attempt->timefinish, $this->strtimeformat);
+            return $attempt->timefinish;
         } else {
             return  '-';
         }
@@ -189,7 +189,7 @@
      */
     public function col_duration($attempt) {
         if ($attempt->timefinish) {
-            return format_time($attempt->timefinish - $attempt->timestart);
+            return $attempt->timefinish - $attempt->timestart;
         } else {
             return '-';
         }
