--- /var/www/html/moodle_20160301/mod/scorm/report/interactions/report.php	2015-03-26 15:49:01.000000000 +0900
+++ /var/www/html/moodle_20160301_analytics/mod/scorm/report/interactions/report.php	2017-03-17 19:37:04.000000000 +0900
@@ -235,7 +235,7 @@
                 foreach ($scoes as $sco) {
                     if ($sco->launch != '') {
                         $table->no_sorting('scograde'.$sco->id);
-                    }
+		    }
                 }
 
                 $table->column_class('picture', 'picture');
@@ -418,9 +418,9 @@
                         $row[] = $OUTPUT->user_picture($user, array('courseid'=>$course->id));
                     }
                     if (!$download) {
-                        $row[] = '<a href="'.$CFG->wwwroot.'/user/view.php?id='.$scouser->userid.'&amp;course='.$course->id.'">'.$scouser->username.'</a>';
+                        $row[] = '<a href="'.$CFG->wwwroot.'/user/view.php?id='.$scouser->userid.'&amp;course='.$course->id.'">'.$scouser->userid.'</a>';
                     } else {
-                        $row[] = $scouser->username;
+                        $row[] = $scouser->userid;
                     }
                     if (empty($timetracks->start)) {
                         $row[] = '-';
@@ -437,12 +437,12 @@
                         if ($download =='ODS' || $download =='Excel' ) {
                             $row[] = userdate($timetracks->start, get_string("strftimedatetime", "langconfig"));
                         } else {
-                            $row[] = userdate($timetracks->start);
+                            $row[] = $timetracks->start;
                         }
                         if ($download =='ODS' || $download =='Excel' ) {
                             $row[] = userdate($timetracks->finish, get_string('strftimedatetime', 'langconfig'));
                         } else {
-                            $row[] = userdate($timetracks->finish);
+                            $row[] = $timetracks->finish;
                         }
                         $row[] = scorm_grade_user_attempt($scorm, $scouser->userid, $scouser->attempt);
                     }
@@ -468,9 +468,17 @@
                                 if (!$download) {
                                     $row[] = '<img src="'.$OUTPUT->pix_url($trackdata->status, 'scorm').'" alt="'.$strstatus.'" title="'.$strstatus.'" /><br/>
                                             <a href="'.$CFG->wwwroot.'/mod/scorm/report/userreporttracks.php?id='.$cm->id.'&amp;scoid='.$sco->id.'&amp;user='.$scouser->userid.
-                                            '&amp;attempt='.$scouser->attempt.'" title="'.get_string('details', 'scorm').'">'.$score.'</a>';
+                                            '&amp;attempt='.$scouser->attempt.'" title="'.get_string('details', 'scorm').'">'. formatDuration($trackdata->total_time).'</a>';
+				    if($sco === end($scoes)){ // add start
+                                    $row[] = '<img src="'.$OUTPUT->pix_url($trackdata->status, 'scorm').'" alt="'.$strstatus.'" title="'.$strstatus.'" /><br/>
+                                            <a href="'.$CFG->wwwroot.'/mod/scorm/report/userreporttracks.php?id='.$cm->id.'&amp;scoid='.$sco->id.'&amp;user='.$scouser->userid.
+                                            '&amp;attempt='.$scouser->attempt.'" title="'.get_string('details', 'scorm').'">'. $score.'</a>';
+					    } // add end
                                 } else {
-                                    $row[] = $score;
+				      $row[] = formatDuration($trackdata->total_time);
+				      if($sco === end($scoes)){ // add start
+				      	      $row[] = $score;
+				      } // add end
                                 }
                                 // interaction data
                                 for ($i=0; $i < $questioncount; $i++) {
@@ -479,7 +487,7 @@
                                         if (isset($trackdata->$element)) {
                                             $row[] = s($trackdata->$element);
                                         } else {
-                                            $row[] = '&nbsp;';
+                                            //$row[] = '&nbsp;';
                                         }
                                     }
                                     if ($displayoptions['resp']) {
@@ -487,7 +495,7 @@
                                         if (isset($trackdata->$element)) {
                                             $row[] = s($trackdata->$element);
                                         } else {
-                                            $row[] = '&nbsp;';
+                                            //$row[] = '&nbsp;';
                                         }
                                     }
                                     if ($displayoptions['right']) {
@@ -505,14 +513,15 @@
                                             }
                                             $row[] = $rightans;
                                         } else {
-                                            $row[] = '&nbsp;';
+                                            //$row[] = '&nbsp;';
                                         }
                                     }
                                 }
                             //---end of interaction data*/
                             } else {
                                 // if we don't have track data, we haven't attempted yet
-                                $strstatus = get_string('notattempted', 'scorm');
+                                //$strstatus = get_string('notattempted', 'scorm');
+                                $strstatus = '';
                                 if (!$download) {
                                     $row[] = '<img src="'.$OUTPUT->pix_url('notattempted', 'scorm').'" alt="'.$strstatus.'" title="'.$strstatus.'" /><br/>'.$strstatus;
                                 } else {
@@ -520,7 +529,7 @@
                                 }
                                 // complete the empty cells
                                 for ($i=0; $i < count($columns) - $nbmaincolumns; $i++) {
-                                    $row[] = '&nbsp;';
+                                    //$row[] = '&nbsp;';
                                 }
                             }
                         }
@@ -602,4 +611,5 @@
             echo $OUTPUT->notification(get_string('noactivity', 'scorm'));
         }
     }// function ends
+
 }
